pkgs/servers/matrix-synapse/matrix-appservice-irc/update.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash -p nodePackages.node2nix nodejs-12_x curl jq nix-update
+
+set -euo pipefail
+# cd to the folder containing this script
+cd $(dirname $0)
+
+CURRENT_VERSION=$(jq -r '.[0]."matrix-appservice-irc" | split("#") | .[1]' package.json)
+TARGET_VERSION=$(curl https://api.github.com/repos/matrix-org/matrix-appservice-irc/releases/latest | jq -r ".tag_name")
+
+if [[ "$CURRENT_VERSION" == "$TARGET_VERSION" ]]; then
+    echo "matrix-appservice-irc is up-to-date: ${CURRENT_VERSION}"
+    exit 0
+fi
+
+echo "matrix-appservice-irc: $CURRENT_VERSION -> $TARGET_VERSION"
+
+cat << EOF > package.json
+[
+    {"matrix-appservice-irc": "git+https://github.com/matrix-org/matrix-appservice-irc.git#$TARGET_VERSION" }
+]
+EOF
```

Comment:
```suggestion
sed -i "s/#$CURRENT_VERSION/#$TARGET_VERSION/" package.json
```

Reply:
`sed` with untrusted input (`$TARGET_VERSION`) is generally a bad idea, because sed escaping is a nightmare. `substituteInplace` would do the trick though.

Reply:
I think we can be pretty sure that TARGET_VERSION is some valid ansi string.

Reply:
Another alternative with JQ:
``` console
$ jq -n --arg TARGET_VERSION "1.10.1" '[{"matrice-appservice-irc": "git+https://github.com/matrix-org/matrix-appservice-irc.git#\($TARGET_VERSION)"}]'
[
  {
    "matrice-appservice-irc": "git+https://github.com/matrix-org/matrix-appservice-irc.git#1.10.1"
  }
]
```